spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/observation/ObservationAutoConfiguration.java

```diff
@@ -90,8 +88,16 @@ OnlyMetricsObservationHandlerGrouping onlyMetricsObservationHandlerGrouping() {
 	}
 
 	@Configuration(proxyBeanMethods = false)
-	@ConditionalOnClass(TracingObservationHandler.class)
-	static class TracingConfiguration {
+	@ConditionalOnBean(Tracer.class)
+	static class MetricsWithTracingConfiguration {
+
+		@Bean
+		@ConditionalOnMissingBean(MeterObservationHandler.class)
+		@ConditionalOnBean(MeterRegistry.class)
+		TracingAwareMeterObservationHandler<Observation.Context> tracingAwareMeterObservationHandler(
+				MeterRegistry meterRegistry, Tracer tracer) {
+			return new TracingAwareMeterObservationHandler<>(new DefaultMeterObservationHandler(meterRegistry), tracer);
+		}
 
 		@Bean
 		TracingObservationHandlerGrouping tracingObservationHandlerGrouping() {
```

Comment:
I think this may need to move. It was previously auto-configured when tracing alone was on the classpath. To be able to auto-configure `TracingAwareMeterObservationHandler` we need both tracing and metrics. Without metrics we'll lose `TracingObservationHandlerGrouping` which I don't think is right.